src/lib/optimizer/strategy/disjunction_to_union_rule.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "abstract_rule.hpp"
+
+namespace opossum {
+
+/**
+ * This rule turns PredicateNodes with disjunctive chains (e.g., `PredicateNode(a OR b OR c)`) as their scan expression
+ *    into n-1 consecutive UnionNodes that combine the results of n PredicateNodes with simple expressions.
+ *    This rule is similar to the PredicateSplitUpRule, which handles ANDs instead of ORs. The DisjunctionToUnionRule
+ *    runs after the PredicateSplitUpRule, so that expressions that are not in conjunctive normal form (ANDs of ORs)
+ *    cannot be fully handled right now. Because the PredicateSplitUpRule already ran, the DisjunctionToUnionRule
+ *    assumes that no ANDs are left in the PredicateExpressions.
```

Comment:
"assumes that no ANDs are left in the PredicateExpressions" - not necessarily true. We might have an AND somewhere deeper within the expressions.

Suggestion: 
> The DisjunctionToUnionRule runs after the PredicateSplitUpRule, so when flattening the top-level logical expression, we expect one or more top-level expressions that were connected using a disjunction ("or"). Conjunctions ("and") might occur nested within such a top-level expression. As the PredicateSplitUpRule already ran, they will not be split up into successive PredicateNodes. That might be a thing for the future, if it ever comes up.